WordPress/openverse
# Break `docker-compose.yml` into smaller `compose.yml` files

## Fixes



Fixes #4111 by @dhruvkb

## Description



This PR breaks down the main `docker-compose.yml` file into smaller `compose.yml` files placed in each of the stack folders. The shared services (like databases) and services not 100% related to any stack (like Plausible stuff) have been retained in the main file.

Additionally the ingestion server and indexer worker also follow the extension approach used by the Airflow services.

The ingestion server can no longer copy over the `docker-compose.yml` file and make edits to it, so it now uses `docker compose config` to let Docker Compose compile the configs (process `include`-d files, interpolate environment variables, normal all configs to the long form etc.) and uses that. This removes a few steps from the compose generation step.

## Testing Instructions


CI checks are passing so that should indicate that Docker Compose services are functional for the most part.

0. Run `just up` and `just ps` to see the state of the services.
1. Check out the repo.
2. Build all the images `just dc build`.
3. Run all the services `just dc up --force-recreate`.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
